src/export/bops/tests/files.test.ts

```diff
@@ -0,0 +1,235 @@
+import { extractTagsFromPassportKey, getBOPSParams } from "../index";
+import type { FileTag } from "../model";
+
+const PASSPORT_UPLOAD_KEY = "proposal.drawing.locationPlan";
+
+const mockFlow = {
+  _root: {
+    edges: ["FnyZh5IrV3"],
+  },
+  FnyZh5IrV3: {
+    data: {
+      fn: "property.drawing.elevation",
+      color: "#EFEFEF",
+    },
+    type: 140,
+  },
+};
+
+const mockBreadcrumbs = {
+  FnyZh5IrV3: {
+    auto: false,
+    data: {
+      "property.drawing.elevation": [
+        {
+          url: "http://localhost:7002/file/private/y2uubi9x/placeholder.png",
+          filename: "placeholder.png",
+          cachedSlot: {
+            file: {
+              path: "placeholder.png",
+              type: "image/png",
+              size: 6146,
+            },
+            status: "success",
+            progress: 1,
+            id: "oPd5GUV_T-bWZWJb0wGs8",
+            url: "http://localhost:7002/file/private/y2uubi9x/placeholder.png",
+          },
+        },
+      ],
+    },
+  },
+};
+
+const mockPassport = {
+  data: {
+    "property.drawing.elevation": [
+      {
+        url: "http://localhost:7002/file/private/y2uubi9x/placeholder.png",
+        filename: "placeholder.png",
+        cachedSlot: {
+          file: {
+            path: "placeholder.png",
+            type: "image/png",
+            size: 6146,
+          },
+          status: "success",
+          progress: 1,
+          id: "oPd5GUV_T-bWZWJb0wGs8",
+          url: "http://localhost:7002/file/private/y2uubi9x/placeholder.png",
+        },
+      },
+    ],
+  },
+};
+
+test("makes file object", () => {
+  const actual = getBOPSParams({
+    breadcrumbs: mockBreadcrumbs,
+    flow: mockFlow,
+    passport: mockPassport,
+    sessionId: "123",
+    flowName: "Apply for a lawful development certificate",
+  }).files;
+
+  const expected = [
+    expect.objectContaining({
+      filename: "http://localhost:7002/file/private/y2uubi9x/placeholder.png",
+    }),
+  ];
+
+  expect(actual).toEqual(expected);
+});
+
+interface TestTag {
+  key: string;
+  tags: Array<FileTag>;
+}
+
+// Google Sheet listing tags: https://bit.ly/3yIscgc
+describe("It extracts tags for", () => {
+  const data: Record<string, TestTag> = {
+    "No passport key": {
+      key: "",
+      tags: [],
+    },
+    "Unmatchable passport key": {
+      key: "foo",
+      tags: [],
+    },
+    "Location plan": {
+      key:
+        PASSPORT_UPLOAD_KEY === "proposal.drawing.locationPlan"
+          ? PASSPORT_UPLOAD_KEY
+          : "key changed unexpectedly!",
+      tags: ["Proposed", /*"Drawing",*/ "Site", "Plan"],
```

Comment:
These are old iterations of accepted BOPS tags; we still hold old submissions that used these tags. Flags were changing frequently for a period of time which I think explains why we were commenting rather than deleting, will flag as a future cleanup task :bookmark: